frontend/src/components/SocialButton.tsx

```diff
@@ -0,0 +1,51 @@
+
+import React from 'react';
+import {Text, TouchableOpacity, View, StyleSheet} from 'react-native';
+
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+const SocialButton = ({buttonTitle, btnType, color, backgroundColor, ...rest}) => {
+  let bgColor = backgroundColor;
+  return (
+    <TouchableOpacity
+      style={[styles.buttonContainer, {backgroundColor: bgColor}]}
+      {...rest}>
+      <View style={styles.iconWrapper}>
+        <FontAwesome name={btnType} style={styles.icon} size={22} color={color} />
+      </View>
+      <View style={styles.btnTxtWrapper}>
+        <Text style={[styles.buttonText, {color: color}]}>{buttonTitle}</Text>
+      </View>
+    </TouchableOpacity>
+  );
+};
+
+export default SocialButton;
+
+const styles = StyleSheet.create({
+  buttonContainer: {
+    marginTop: 10,
+    width: '100%',
+    height: 70,
+    padding: 10,
+    flexDirection: 'row',
+    borderRadius: 3,
+  },
+  iconWrapper: {
+    width: 30,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  icon: {
+    fontWeight: 'bold',
+  },
+  btnTxtWrapper: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  buttonText: {
+    fontSize: 18,
+    fontWeight: 'bold',
+  },
+});
```

Comment:
We _shuold_ end with a blank line - it's an old linux thing - should be fine nowadays, but to be safe